docs/design-proposals/multi-tenancy/multi-tenancy-network-mizar-integration.md

```diff
@@ -0,0 +1,201 @@
+---
+title: Arktos Multi-Tenancy Network Based on Mizar
+authors:
+- "@sindica"
+---
+
+# Arktos Multi-Tenancy Network Based on Mizar
+
+## Table of Content
+1. [Introduction](#intro)
+2. [Goals](#goal)
+3. [Proposal](#proposal)<br>
+   3.1. [Related Mizar Specs](#mizar-spec)<br>
+   3.2. [Related Arktos Specs](#arktos-spec)<br>
+   3.3. [Arktos Pod Creation Workflow](#pod-creation-wf)<br>
+   &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;3.3.1 [Workflow for default VPC and subnet in tenant](#pod-creation-default-wf)<br>
+   &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;3.3.2 [Workflow for secondary VPC and subnet in tenant](#pod-creation-second-vpc-wf)<br>
+   &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;3.3.3 [Workflow for secondary subnet in VPC](#pod-creation-second-subnet-wf)<br>
+
+## 1. Introduction <a name="intro"></a>
+
+This document is a part of the multi-tenancy network design document that focuses on how to integrated [Mizar network solution](https://github.com/CentaurusInfra/mizar) into Arktos.
+
+The generic network design is in [Multi-Tenancy Network](multi-tenancy-network.md).
+
+## 2. Goals <a name="goal"></a>
+
+* Primary: support multi-tenancy communication based on existing Mizar network implementation. Multi-tenancy communication 
+includes connectivity amongs pods in the same tenant, as well as network isolation across different tenants.
+* Primary: upon tenant creation, automatically create Mizar VPC and subnet for future communication within this tenant.
+* Primary: upon tenant pod creation, automatically attach pod to Mizar VPC/subnet created for this tenant.
+* Primary: Mizar and Arktos decouple, i.e. Mizar manage VPCs and subnets in VPC only, it should not be aware of the ownership
+of different pods and other objects.
+* Secondary: support multiple subnets in same VPC for a single tenant. (Post 2022-01-30)
+* Secondary: support multiple VPC/subnets in same tenant. (Post 2022-01-30)
+
+## 3. Proposal <a name="proposal"></a>
+
+Inject Mizar adapter into Arktos, completely isolated from Mizar management plane and relatively isolated from Arktos network
```

Comment:
is "adapter" to be the formal team in Arktos ? we use this a lot in scrum but officially they should be termed as "controllers".

Reply:
There are mizar node/service/pod controller in addition to original node/service/pod controller. Even though mizar controllers are currently cohosted together with other controllers in KCM, that list/watch arktos objects and making api calls to mizar services. Those arktos objects already has their own controllers for non mizar related tasks. I feel adapter is a proper name for the mizar related controllers.

Reply:
that could be OK if this is the case. i would suggest to elaborate it a bit as : currently we use Arktos control plane as Mizar's control plane for simplicity reason, so those set of "controllers" can be considered as "adaptors".